## HTS Analysis Repository

A home for R, mothur, and some python based solutions for analysing your HTS datasets.

### Tutorials

Follow these standard workflows for a jumping off point in your HTS dataset analysis.

- [FastQ to Mothur](https://mgaley-004.github.io/MiSeq-Analysis/Tutorials/fastQ2Mothur.html)
- Data Cleanup and Organization in R
- Rarefactions and Diversity
- [Ordinations and Redundancy Analysis](https://mgaley-004.github.io/MiSeq-Analysis/Tutorials/ordinationRDA.html)
- Describing Major Lineages
- [FAPROTAX and Functions](https://mgaley-004.github.io/MiSeq-Analysis/Tutorials/faprotax_demo.html)
- [Source Sink Dynamics](https://mgaley-004.github.io/MiSeq-Analysis/Tutorials/SourceSink.html)


### General Help

- [Installing Python](https://mgaley-004.github.io/MiSeq-Analysis/Help/Python.html)
- [SLURM computing](https://mgaley-004.github.io/MiSeq-Analysis/Tutorials/SLURM/SlurmTransition.html)[(PDF)](https://mgaley-004.github.io/MiSeq-Analysis/Help/meeting121820.pdf)

### Other Helpful Packages and Tutorials

- Having trouble in mothur? Check the [SOP](https://mothur.org/wiki/miseq_sop/) and the [wiki](https://mothur.org/wiki/mothur_manual/)
- Compositional analysis is complicated. The ALDEx2 package, used here frequently, is really helpful and the Gloor lab has extensive documentation and a helpful [workshop](https://github.com/ggloor/CoDa_microbiome_tutorial/wiki)
- I'm a big fan of [mixOmics](http://mixomics.org/). For fancier PCA options, check it out.


### Helpful Publications

- Miranda's long list of favorite bioinformatics articles

### Authors
Miranda Galey [@mgaley-004](https://github.com/mgaley-004/)
